### PR TITLE
Add public for ->writeToDisk() as option

### PR DESCRIPTION
**Pull request by bmckay959**
I'm using S3 for hosting my sitemaps(Using Vapor for the site so local disk isn't an option).

I'm currently generating the sitemap and after creation, I'm then changing the visibility to public. This could cause, though rare, a race condition of me creating the sitemap and before making it public, a search engine crawler would receive a "Access Denied" error from S3.

By adding the visibility option as seen here(https://laravel.com/docs/10.x/filesystem#file-visibility), it resolves this issue.

### CODE REVIEW

File: src/Sitemap.php [hunk 1]
```diff
@@ -78,9 +78,15 @@ public function writeToFile(string $path): static
         return $this;
     }
 
-    public function writeToDisk(string $disk, string $path): static
+    public function writeToDisk(string $disk, string $path, bool $public = false): static
     {
-        Storage::disk($disk)->put($path, $this->render());
+        if($public) {
```
Comment (freekmurze) [inline src/Sitemap.php:83]: Could you refactor this to a ternary?

**Review by freekmurze — CHANGES_REQUESTED** [review]
Could you add tests and docs for this change?

**Comment by bmckay959** [comment]
I made the if statements ternary....I'm trying to work through the testing here. `Storage::fake()` uses local storage from what I can tell, which makes the `Storage::getVisibility()` return `League\Flysystem\UnableToRetrieveMetadata: Unable to retrieve the visibility for file at location: sitemap.xml` since it's not utilizing the aws driver. I'll try to look into it more tomorrow/over the weekend when I get time.

**Comment by bmckay959** [comment]
Just so you can see what I tried:

```php
it('can write a sitemap to a storage disk and make visibility be public', function () {
    Storage::fake('s3');
    $this->index->writeToDisk('s3', 'sitemap.xml', true);

    $visibility = Storage::getVisibility('sitemap.xml');

    assertMatchesXmlSnapshot(Storage::disk('s3')->get('sitemap.xml'));
    expect($visibility)->toBe('public');
});
```

**Comment by freekmurze** [comment]
Thank you!

**Comment by ismaail** [comment]
google search console was giving error `Sitemap could not be read` because the

**Pull request merged**